File: src/org/labrad/qubits/mem/SendFiberCommand.java
package org.labrad.qubits.mem;

import org.labrad.qubits.enums.DacFiberId;

public class SendFiberCommand implements MemoryCommand {
  private DacFiberId channel;
  private int bits;

  public SendFiberCommand(DacFiberId channel, int bits) {
    this.channel = channel;
    this.bits = bits;
  }

  public long[] getBits() {
    int send;
    switch (channel) {
      case FOUT_0: send = 0x100000; break;
      case FOUT_1: send = 0x200000; break;
      default:
        throw new RuntimeException("Invalid DAC fiber id: " + channel);
    }
    return new long[] {send + (bits & 0x0FFFFF)};
  }
}

File: QubitServer/src/main/java/org/labrad/qubits/mem/SendFiberCommand.scala
package org.labrad.qubits.mem

import org.labrad.qubits.FpgaModelDac
import org.labrad.qubits.enums.DacFiberId

case class SendFiberCommand(channel: DacFiberId, bits: Int) extends MemoryCommand {

  def getBits(): Array[Long] = {
    val send = channel match {
      case DacFiberId.FOUT_0 => 0x100000
      case DacFiberId.FOUT_1 => 0x200000
      case DacFiberId.FIN => sys.error(s"Invalid DAC fiber id: $channel")
    }
    Array[Long](send + (bits & 0x0FFFFF))
  }

  def getTime_us(dac: FpgaModelDac): Double = {
    FpgaModelDac.clocksToMicroseconds(1)
  }
}
